matrix.go

```diff
@@ -342,3 +343,37 @@ func (mx *WebsocketCommandHandler) HandleSyncProxyError(syncErr *mautrix.RespErr
 	atomic.StoreInt64(&mx.syncProxyWaiting, 0)
 	mx.bridge.RequestStartSync()
 }
+
+type NewRoomBackfillRequest struct {
+	Chats []*imessage.ChatInfo `json:"chats"`
+}
+
+type NewRoomBackfillResponse map[string]id.RoomID
+
+func (mx *WebsocketCommandHandler) handleCreateRoomsForBackfill(cmd appservice.WebsocketCommand) (bool, any) {
+	var req NewRoomBackfillRequest
+	if err := json.Unmarshal(cmd.Data, &req); err != nil {
+		return false, fmt.Errorf("failed to parse request: %w", err)
+	}
+
+	createdRooms := NewRoomBackfillResponse{}
+
+	for _, info := range req.Chats {
+		info.Identifier = imessage.ParseIdentifier(info.JSONChatGUID)
+		portal := mx.bridge.GetPortalByGUID(info.Identifier.String())
+
+		if len(portal.MXID) > 0 {
+			portal.zlog.Info().Msg("Syncing Matrix room with latest chat info")
+			portal.SyncWithInfo(info)
+		} else {
+			portal.zlog.Info().Msg("Creating Matrix room with latest chat info")
+			err := portal.CreateMatrixRoom(info, nil)
+			if err != nil {
+				return false, fmt.Errorf("failed to create portal: %w", err)
+			}
+		}
```

Comment:
I'm not sure if I need to sync if the portal has already been created... I could just call `CreateMatrixRoom` unconditionally and it will just be a no-op if the room already exists for the portal.